week2/homework/src/index.js

```diff
@@ -1,3 +1,161 @@
+  remove(todoItem);
+} else if (command === 'update') {
+  update(todoItem, newVal);
+} else if (command === 'reset') {
+  reset();
+} else if (command === 'help' || !command) {
+  help();
+}
 
-// TODO: Write the homework code in this file
+function list() {
+  fs.readFile('./src/todoList.txt', 'utf8', (error, todoList) => {
+    if (error) {
+      if (error.code === 'ENOENT') {
+        console.log('no data found');
+      } else {
+        console.log(error);
+      }
+    } else {
+      console.log(todoList);
+    }
+  });
+}
+function add(todoItem) {
+  fs.appendFile('./src/todoList.txt', todoItem + '\n', error => {
+    // appendFile: creates the file if it does not exist already
+    if (error) {
+      console.error(error);
+    } else {
+      console.log(`[${todoItem}] was added`);
+    }
+  });
+}
+/* remove()
+1. Read existing todoList.txt
+2. Filter the item to be removed
+3. Write new contents back to todoList.txt */
+function remove(num, arr) {
+  fs.readFile('./src/todoList.txt', 'utf8', (error, todoList) => {
+    if (error) {
+      if (error.code === 'ENOENT') {
+        console.log('no data found');
+      } else {
+        console.log(error);
+      }
+    } else {
+      if (num < 0 || num > todoList.length) {
+        console.log('Please write a valid number');
+      } else {
+        const splitList = todoList.split('\n');
+        // make the list an array of lines
+        const newList = splitList.filter((elem, index) => index !== num - 1).join('\n');
+        console.log(newList);
+        fs.writeFile('./src/todoList.txt', newList, error => console.error(error));
+      }
+    }
+  });
+}
+/* update()
+1. Read existing todoList.txt
+2. Map all the items to update the item
+3. Write new contents back to todoList.txt */
+function update(num, newVal, arr) {
+  fs.readFile('./src/todoList.txt', 'utf8', (error, todoList) => {
+    if (error) {
+      if (error.code === 'ENOENT') {
+        console.log('no data found');
+      } else {
+        console.log(error);
+      }
+    } else {
+      if (num < 0 || num > todoList.length) {
+        console.log('Please write a valid number');
+      } else {
+        const splitList = todoList.split('\n');
+        console.log(num);
+        console.log(newVal);
+
+        // make the list an array of lines
+        const newList = splitList
+          .map((elem, index) => (index === num - 1 ? newVal : elem))
+          .join('\n');
+        // console.log(newList);
+        fs.writeFile('./src/todoList.txt', newList, error => console.error(error));
+      }
+    }
+  });
+}
+function reset() {
+  fs.writeFile('./src/todoList.txt', '', (error, todoList) => {
+    if (error) {
+      if (error.code === 'ENOENT') {
+        console.log('no data found');
+      } else {
+        console.log(error);
+      }
+    } else {
+      console.log(`Rest!
+      Nothing to be done in the list`);
+    }
+  });
+}
+function help() {
+  fs.readFile('./src/todoList.txt', 'utf8', error => {
```

Comment:
Help command is only for printing the usage guide, nothing else. So no need to check the existence of the data file.